src/youtube.js

```diff
@@ -0,0 +1,1094 @@
+        bandwidth: 0,
+        width: i+1, //TODO: get estimated width and height from https://developers.google.com/youtube/iframe_api_reference#Playback_quality
+        height: i+1,
+        language: '',
+        index: i
+      };
+      videoTracks.push(new VideoTrack(settings));
+    });
+    return videoTracks;
+  }
+
+  /**
+   * Set a source.
+   * @param {string} source - Source to set.
+   * @public
+   * @returns {void}
+   */
+  set src(source: string): void {
+    this._source.url = source;
+  }
+
+  /**
+   * Get the source url.
+   * @returns {string} - The source url.
+   * @public
+   */
+  get src(): string {
+    return (this._loaded && this._source) ? this._source.url: "";
+  }
+
+  /**
+   * Get the current time in seconds.
+   * @returns {Number} - The current playback time.
+   * @public
+   */
+  get currentTime(): number {
+    return this._api ? this._api.getCurrentTime() : 0;
+  }
+
+  /**
+   * Set the current time in seconds.
+   * @param {Number} to - The number to set in seconds.
+   * @public
+   * @returns {void}
+   */
+  set currentTime(to: number): void {
+    if (this._api) {
+      if (this.currentTime !== to) {
+        this._isSeeking = true;
+        this._api.seekTo(to, true);
+        this.dispatchEvent(new FakeEvent(EventType.SEEKING));
+        if (this.paused) {
+          this._startSeekTargetWatchDog();
+        }
+      }
+    }
+  }
+
+  _startSeekTargetWatchDog() {
+    const previousTime = this.currentTime;
+    this._stopSeekTargetWatchDog();
+    this._seekTargetIntervalId = setInterval(() => {
+      if (!this.paused || !this._isSeeking) {
+        this._stopSeekTargetWatchDog();
+      } else if (this.currentTime !== previousTime) {
+        this.dispatchEvent(new FakeEvent(EventType.TIME_UPDATE));
+        this.dispatchEvent(new FakeEvent(EventType.SEEKED));
+        this._isSeeking = false;
+      }
+    }, TIME_INTERVAL_TIMEOUT);
+  }
+
+  _stopSeekTargetWatchDog() {
+    clearInterval(this._seekTargetIntervalId);
+    this._seekTargetIntervalId = null;
+  }
+
+  /**
+   * Get the duration in seconds.
+   * @returns {Number} - The playback duration.
+   * @public
+   */
+  get duration(): number {
+    return this._api ? this._api.getDuration() : NaN;
+  }
+
+  /**
+   * Set playback volume.
+   * @param {Number} vol - The volume to set.
+   * @public
+   * @returns {void}
+   */
+  set volume(vol: number): void {
+    this._api && this._api.setVolume(vol * 100);
+  }
+
+  /**
+   * Get playback volume.
+   * @returns {Number} - The volume value of the video element.
+   * @public
+   */
+  get volume(): number {
+    return this._api.getVolume() / 100;
+  }
+
+  ready() {}
+
+  /**
+   * Get paused state.
+   * @returns {boolean} - The paused value of the video element.
+   * @public
+   */
+  get paused(): boolean {
+    if (this._api && this._api.getPlayerState) {
+      return ![window.YT.PlayerState.PLAYING, window.YT.PlayerState.BUFFERING].includes(this._api.getPlayerState());
```

Comment:
YT is nullable no?

Reply:
If `_this._api` exist it means Iframe API was loaded